great_expectations/expectations/metrics/util.py

```diff
@@ -703,6 +703,23 @@ def _get_normalized_column_name_mapping_if_exists(
             ) or (column_name == str(typed_column_name_cursor)):
                 return column_name, typed_column_name_cursor
 
+            # use explicit identifier if passed in by user
+            if (type(typed_column_name_cursor) == str) and (
+                (
+                    column_name.casefold().strip('"')
+                    == typed_column_name_cursor.casefold()
+                )
+                or (
+                    column_name.casefold().strip("[]")
+                    == typed_column_name_cursor.casefold()
```

Comment:
Today I learned about using the `casefold()` method over `lower()`. Thank you @TrangPham

Reply:
@TrangPham Why would we be looking for `"` and/or "[]"?  Are these delimiters in the specifications?  Interesting problem, by the way.  I know that we are dealing with Spark here, but it is also quite possible that the use of `casefold()` for SQLAlchemy is not a perfect solution either (I heard from a user that they still had problems with Snowflake, but they did not provide an example; other problems reported with Snowflake were fixed as far as I can tell).  Thanks.

Reply:
I would prefer to be able to deduce the delimiters from the dialect if possible. I was playing around with SQLalchemy and quoted_name but could not figure out an elegant solution. The code is passed through many partials as well which makes it hard to understand where the engine should come from. 

I wasn't sure if using quoted_name to standardize to would result in a correctly delimited identifier.

```
def _verify_column_names_exist_and_get_normalized_typed_column_names_map(
    column_names: List[str] | str,
    batch_columns_list: List[str | sqlalchemy.quoted_name],
    error_message_template: str = 'Error: The column "{column_name:s}" in BatchData does not exist.',
    verify_only: bool = False,
) -> List[Tuple[str, str | sqlalchemy.quoted_name]] | None:
    """
    Insures that column name or column names (supplied as argument using "str" representation) exist in "Batch" object.

    Args:
        column_names: Single string-valued column name or list of string-valued column names
        batch_columns_list: Properly typed column names (output of "table.columns" metric)
        verify_only: Perform verification only (do not return normalized typed column names)
        error_message_template: String template to output error message if any column cannot be found in "Batch" object.

    Returns:
        List of tuples having mapping from string-valued column name to typed column name; None if "verify_only" is set.
    """
```

Would calling this function with `batch_columns_list` as a list of quoted_name result in the correct raw SQL for the dialect be used?

Reply:
@alexsherstinsky @Shinnnyshinshin 

I played around with this more

I'm not sure how quoted_name functionality works/would help
<img width="1686" alt="Screenshot 2023-06-01 at 5 11 34 PM" src="https://github.com/great-expectations/great_expectations/assets/2117313/7e050e7e-2c40-4bf8-a049-a3d256aa4d94">

But I think we could use `Column` class to force quoting on all columns and normalize to that
<img width="1910" alt="Screenshot 2023-06-01 at 4 59 45 PM" src="https://github.com/great-expectations/great_expectations/assets/2117313/6fc6a819-81c5-4f2a-afe5-45dc05a12059">

With current code line 158 would fail comparing "`other names" with "other names".
If "other names" is passed in, line 158 would not fail but the raw SQL would be incorrect.

I like the idea of either:
1. forcing user to use explicit identifiers if they want it
2. OR allowing user to pass in any string and always forcing quotes for them